pkg/builder/interface.go

```diff
@@ -70,6 +75,25 @@ func IsDone(status *v1alpha1.BuildStatus) bool {
 	return false
 }
 
+// IsTimedOut returns true if the build's status indicates that build is timedout.
+func IsTimeout(status *v1alpha1.BuildStatus, buildTimeout string) bool {
+	defaultTimeout := "10m"
```

Comment:
It might be simpler to have the default already stated in terms of a `time.Duration`:

```go
var timeout time.Duration
if buildTimeout == "" {
  timeout = 10*time.Minute
} else {
  timeout, err = time.ParseDuration(buildTimeout)
  if err != nil {
    return err
  }
}
```